### PR TITLE
Feat: Expose optimizeDisableComponentRevision in helm chart

### LINKED ISSUE #3459: Expose optimize configuration in helm chart
### Discussed in https://github.com/oam-dev/kubevela/discussions/3448

<div type='discussions-op-text'>

<sup>Originally posted by **stigtermichiel** March 15, 2022</sup>
As referenced in this issue: https://github.com/oam-dev/kubevela/issues/3296, it was recommended to set `--optimize-disable-component-revision`. 

We deploy vela-core with helm and saw that this flag is not exposed in the chart. Is there another way of doing it?</div>

Here's an example about how to expose it:

1. add the args in controller:  https://github.com/oam-dev/kubevela/blob/master/charts/vela-core/templates/kubevela-controller.yaml#L141
2. add that in chart https://github.com/oam-dev/kubevela/blob/master/charts/vela-core/values.yaml#L47
3. add explanation https://github.com/oam-dev/kubevela/blob/master/charts/vela-core/values.yaml#L40

We should add an `optimize` section for that.

### PR DESCRIPTION
**Pull request by stigtermichiel**
### Description of your changes

Fixes #3459

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

Installed chart locally.

### CODE REVIEW

**Comment by msftgits** [comment]
[CLA assistant check](https://cla.opensource.microsoft.com/oam-dev/kubevela?pullRequest=3470) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/oam-dev/kubevela?pullRequest=3470) before we can accept your contribution.<br/><br/>:x: stigtermichiel [sign now](https://cla.opensource.microsoft.com/oam-dev/kubevela?pullRequest=3470)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/oam-dev/kubevela?pullRequest=3470) it.</sub>

**Comment by stigtermichiel** [comment]
I can do the other optimizations as well if you have a chance to look at this today. Otherwise, I will be on holiday next week :)

**Review by wonderflow — COMMENTED** [review]
LGTM， please also finish the others

File: charts/vela-core/values.yaml [hunk 1]
```diff
@@ -84,6 +84,10 @@ webhookService:
 healthCheck:
   port: 9440
 
+## @section KubeVela controller optimization parameters
+
+##@param optimizeDisableComponentRevision Optimize componentRevision by disabling the creation and gc
+optimizeDisableComponentRevision: false
```
Comment (wonderflow) [inline charts/vela-core/values.yaml:90]:
```suggestion
optimize:
    disableComponentRevision: false
```

Comment (wonderflow) [reply charts/vela-core/values.yaml:90]: let's make the optimize in a separate section

**Comment by wonderflow** [comment]
@stigtermichiel Are you still working on this? This is useful I think.

**Comment by stigtermichiel** [comment]
Sorry, totally forgot about this. Will finish it today!

**Comment by wonderflow** [comment]
Great, thanks

**Review by a14m — APPROVED** [review]
LGTM

**Review by Somefive — APPROVED** [review]
Great work! Some of the flags might be moved into FeatureFlags part in the future. If you are interested in this part, we can discuss more about it :).

**Pull request merged**